dvc/repo/artifacts.py

```diff
@@ -0,0 +1,67 @@
+import logging
+import os
+import re
+from typing import TYPE_CHECKING, Dict
+
+from dvc.annotations import Artifact
+from dvc.dvcfile import FileMixin
+from dvc.utils import relpath
+
+if TYPE_CHECKING:
+    from dvc.repo import Repo
+
+logger = logging.getLogger(__name__)
+
+
+NAME_RE = re.compile(r"^[a-z][a-z0-9-/.]*[a-z0-9]$")
+
+
+def name_is_compatible(name: str) -> bool:
+    return bool(NAME_RE.search(name))
+
+
+def check_name_format(name: str) -> None:
+    if not name_is_compatible(name):
+        logger.warning(
+            "To be compatible with Git tags and GTO, artifact name ('%s') "
+            "must satisfy the following regex: %s",
```

Comment:
@aguschin Not a blocker, and I'm fine with keeping the restrictions as is, but a couple suggestions on the warning and how we document it:

1. Maybe we should not focus on Git tags (could be confusing since the restrictions aren't the same as what Git allows) or GTO (since it becomes more of a background utility and users may not know what it is). We could include a mention of the registry or leave out the rationale and only state that artifact names should follow this pattern.
2. Listing the regex doesn't seem that user-friendly. I'm not sure it's even a safe assumption that users know regex. My best try to explain the regex is "You can use letters and numbers, and you can use `-`, `/`, and `.` as separators (but not at the start or end). The first character must be a letter, and the minimum length is two characters." The 2-character requirement bothers me most since I can't think of any rationale or way to integrate it with the other rules. It feels like we are optimizing for a simple regex over a simple explanation. 

Some possible tweaks to the regex:
1. `^[a-z]([a-z0-9-/.]*$`, explained as "You can use letters, numbers, `-`, `/`, and `.`. The starting character must be a letter."
2. `^[a-z]([a-z0-9-/.]*[a-z0-9])?$`, explained as "You can use letters and numbers, and you can use `-`, `/`, and `.` as separators (but not at the start or end). The first character must be a letter."

Reply:
Thanks! Used your feedback.